### PR TITLE
Fix: event.origin tag for macOS and other Apple platforms

### LINKED ISSUE #614: `event.origin` tag should be `apple` or `ios` and `macos` instead of `ios` for both platforms
https://github.com/getsentry/sentry-dart/blob/main/flutter/ios/Classes/SentryFlutterPluginApple.swift#L243

https://github.com/getsentry/sentry-docs/pull/4234/files#r723878494

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: event.origin tag for macOS and other Apple platforms

## :bulb: Motivation and Context
Fix https://github.com/getsentry/sentry-dart/issues/614

## :green_heart: How did you test it?
running on iOS and macOS

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] I updated the docs if needed
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: flutter/ios/Classes/SentryFlutterPluginApple.swift [hunk 1]
```diff
@@ -240,7 +240,21 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
 
             switch sdk["name"] as? String {
             case "sentry.cocoa":
-                setEventEnvironmentTag(event: event, origin: "ios", environment: "native")
+                #if os(OSX)
```
Comment (romtsn) [inline flutter/ios/Classes/SentryFlutterPluginApple.swift:243]: Just curious if it's possible to do via switch-case?

Comment (romtsn) [reply flutter/ios/Classes/SentryFlutterPluginApple.swift:243]: Ok, Google said no

**Review by romtsn — APPROVED** [review]

**Comment by bruno-garcia** [comment]
If we get these on the develop docs we can align with other hybrid sdks

**Comment by denrase** [comment]
@marandaneto Was skimming through my GH issues and saw this PR. Does `os(OSX)` work as intended? I have it somewhere in the back of my head that it should be `os(macOS)`, but i couldn't find the documentation for it.

**Comment by marandaneto** [comment]
> @marandaneto Was skimming through my GH issues and saw this PR. Does `os(OSX)` work as intended? I have it somewhere in the back of my head that it should be `os(macOS)`, but i couldn't find the documentation for it.

yes, https://sentry.io/organizations/sentry-sdks/issues/2390892737/?project=5428562&query=is%3Aunresolved

**Pull request merged**